src/libsync/propagateupload.h

```diff
@@ -321,6 +322,100 @@ private slots:
     void slotUploadProgress(qint64,qint64);
 };
 
+/**
+ * @brief The PropagateFiles class is a container class for upload jobs under chunking size.
+ *
+ * It will also ensure proper bandwidth utilization vs bookkeeping balance, and that in case no other items then under chunk uploads are available,
+ * it will parallelise itself into 3 flows.
+ *
+ *
+ * @ingroup libsync
+ *
+ * State Machine:
+ *
+ *  _________________________________________________ _________________________________________
+ * |                                                 |                                         |
+ * |                     (Are there any STANDARD jobs to run in whole sync?)                   |
+ * |                   Yes |                                             | No                  |
+ * |  (Is there any DB job running in whole sync?)   (Is there any DB job within this class?)  |
+ * |    Yes |                       | No                          No |              | Yes      |
+ * |        |                       |                                |              |          |
+ * |<-[Schedule STANDARD job]       |                                |      [Schedule DB job]->
+ * |                                |                                |
+ * |                                |                                |
+ * |       (Is there any DB job within this class?)                  |
+ * ^          Yes |                       | No                       |
+ * |              |                       |                          |
+ * |<-[Schedule DB job]                   |                          |
+ * |                                      |                          |
+ * |                                      |                          |
+ * ^         (Are there at least 2 STANDARD jobs                     |
+ * |          already running within this class?)                    |
+ * |          No  |                      | Yes                       |
+ * |              |                      |                           |
+ *  <-[Schedule STANDARD job]             ->[Try in next container]<-
+ *
+ */
+class PropagateFiles : public PropagatorJob {
```

Comment:
Because PropagateDirectory is much more complicated thinking about cases to handle, did not want to introduce even more complexity, we will get lost. Also I wanted to make this "pluggable" so you can easily switch it on and off.